R/get_vaa.R

```diff
@@ -54,39 +66,94 @@ get_vaa_names <- function() {
 #' get_vaa("slope")
 #' get_vaa(c("slope", "lengthkm"))
 #'
+#' get_vaa(updated_network = TRUE)
+#' get_vaa("reachcode", updated_network = TRUE)
+#'
 #' #cleanup if desired
 #' unlink(dirname(get_vaa_path()), recursive = TRUE)
 #' }
 
 get_vaa <- function(atts = NULL,
                     path = get_vaa_path(),
-                    download = TRUE) {
+                    download = TRUE,
+                    updated_network = FALSE) {
+
+  check_vaa_path(path, download, FALSE)
 
-  check_vaa_path(path, download)
+  if(updated_network) {
+    updated_net_path <- file.path(dirname(path), "enhd_nhdplusatts.fst")
 
-  avaliable_names = get_vaa_names()
+    check_vaa_path(updated_net_path, download, TRUE)
 
-  bad_atts = atts[!atts %in% avaliable_names]
-  atts      = atts[atts %in% avaliable_names]
-  if(length(bad_atts) > 0){
-    message(paste(bad_atts, collapse = ", "), " not in vaa data. Ignoring...")
+    new_names <- fst::metadata_fst(updated_net_path)[["columnNames"]]
   }
 
-  if(is.null(atts)){
-    return(fst::read.fst(path))
+  available_names = get_vaa_names()
+
+  if(is.null(atts)) {
+
+    atts <- available_names
+
+  } else {
+
+    bad_atts = atts[!atts %in% available_names]
+    atts      = atts[atts %in% available_names]
+    if(length(bad_atts) > 0){
+      message(paste(bad_atts, collapse = ", "), " not in vaa data. Ignoring...")
+
+      if(length(atts) == 0) {
+        stop()
+      }
+
+    }
+
   }
 
-  if(all(atts %in% avaliable_names)){
+  if(updated_network) {
+
+    message("Caution: updated attributes drop some catchments and attributes")
```

Comment:
@mikejohnson51 -- wondering what you think here. The new nhdplus attributes invalidate the ones below and don't include coastal flowlines and maybe some others (need to verify which are actually missing).